modules/bench/src/main/clojure/xtdb/bench/yakbench.clj

```diff
@@ -0,0 +1,531 @@
+                  (str "select count(*) as cnt "
+                       "from user_items ui "
+                       "group by ui.user_item$user")
+                  :cnt))
+
+(defn user-subs-stats
+  [conn]
+  (stats-from-sql conn
+                  (str "select count(*) as cnt "
+                       "from subs s "
+                       "group by s.sub$user")
+                  :cnt))
+
+(defn feed-items-stats
+  [conn]
+  (stats-from-sql conn
+                  (str "select count(*) as cnt "
+                       "from items i "
+                       "group by i.item$feed$feed")
+                  :cnt))
+
+(defn inspect
+  [node {{:keys [user-id]} :max-user :keys [active-users]}]
+  (with-open [conn (get-conn node)]
+    (let [tables ["users" "feeds" "subs"
+                  "items" "user_items" "ads"
+                  "ad_credits" "ad_clicks" "digests"
+                  "bulk_sends" "skips"]
+          rows (for [t tables]
+                 {:table t
+                  :count (:xt/column-1 (first (xt/q conn [(str "select count(*) from " t)])))})]
+      (println)
+      (println "Table counts")
+      (pp/print-table [:table :count] rows)
+      (println))
+
+    (let [queries [["subs" "select count(*) from subs where sub$user = ?" [user-id]]
+                   ["items" "select count(*) from items where item$feed$feed is not null and item$feed$feed in (select sub$feed$feed from subs where sub$user = ?)" [user-id]]
+                   ["user_items" "select count(*) from user_items where user_item$user = ?" [user-id]]
+                   ["ad_clicks" "select count(*) from ad_clicks where ad.click$user = ?" [user-id]]
+                   ["digests" "select count(*) from digests where digest$user = ?" [user-id]]
+                   ["skips" "select count(*) from skips where skip$user = ?" [user-id]]
+                   ["ads" "select count(*) from ads where ad$user = ?" [user-id]]
+                   ["ad_credits"
+                    (str "select count(c._id) \n"
+                         "from ad_credits c \n"
+                         "join ads a on c.ad.credit$ad = a._id \n"
+                         "where a.`ad$user` = ?")
+                    [user-id]]]]
+
+      (println "Picked User" (xt/q conn ["select * from users where _id = ?" user-id]))
+      (println)
+      (println "Max User Counts")
+      (let [rows (for [[label sql args] queries]
+                   {:table label
+                    :count (:xt/column-1 (first (xt/q conn (into [sql] args))))})]
+        (pp/print-table [:table :count] rows))
+      (println))
+
+    (println "Active Users" (count active-users))
+    (println)
+
+    (println "User items stats")
+    (pp/print-table [(user-items-stats conn)])
+    (println)
+
+    (println "User subs stats")
+    (pp/print-table [(user-subs-stats conn)])
+    (println)
+
+    (println "Feed items stats")
+    (pp/print-table [(feed-items-stats conn)])
+    (println)))
+
+(defn load-data!
```

Comment:
if this is required to make the dataset loadable at arbitrary scale, let's get this into the dataset ns?